pkg/operator/ceph/cluster/mon/spec.go

```diff
@@ -213,3 +215,29 @@ func (c *Cluster) makeMonDaemonContainer(monConfig *monConfig) v1.Container {
 
 	return container
 }
+
+// UpdateCephDeploymentAndWait verifies a deployment can be stopped or continued
+func UpdateCephDeploymentAndWait(context *clusterd.Context, deployment *apps.Deployment, namespace, clusterName string, cephVersion cephver.CephVersion) error {
```

Comment:
To avoid parsing strings to determine the daemon ID, here is where I would expect the operator to pass in the daemon ID I think.